r/vignettes/data_wrangling.Rmd

```diff
@@ -165,6 +165,34 @@ sw2 %>%
   transmute(name, height, mass, res = residuals(lm(mass ~ height)))
 ```
 
+Because window functions are not supported, computing an aggregation like `mean()` on a grouped table or within a rowwise opertation like `filter()`  is not supported:
+
+```{r}
+sw %>%
+  select(1:4) %>%
+  filter(!is.na(hair_color)) %>%
+  group_by(hair_color) %>%
+  filter(height < mean(height, na.rm = TRUE))
+```
+
+This operation can be accomplished in arrow by computing the aggregation separately, for example within a join operation: 
+
+```{r}
+
+sw %>%  
+  select(1:4) %>%
+  filter(!is.na(hair_color)) %>%
+  left_join(sw %>% 
+    group_by(hair_color) %>%
+    summarize(mean_height = mean(height, na.rm = TRUE)) 
+    ) %>%
+  filter(height < mean_height) %>% 
+  select(-mean_height) %>%
+  collect()
+
```

Comment:
How about formatting it as follows?

```suggestion
sw %>%
  select(1:4) %>%
  filter(!is.na(hair_color)) %>%
  left_join(
    sw %>% 
      group_by(hair_color) %>%
      summarize(mean_height = mean(height, na.rm = TRUE)) 
    ) %>%
  filter(height < mean_height) %>% 
  select(!mean_height) %>%
  collect()
```

Reply:
Sorry for the lack of explanation.
I believe dplyr has been used `!` these days, instead of `-` in select, etc.
The reference page confirms that there are no examples using `-`.
https://dplyr.tidyverse.org/reference/select.html

Reply:
Great, makes sense, and I agree with the change then.